### PR TITLE
Support for running tests locally

### PR DESCRIPTION
**Pull request by jhvhs**
This pull request simplifies running tests with all the dependencies on a local workstation

[#167952380](https://www.pivotaltracker.com/story/show/167952380)

Signed-off-by: Konstantin Semenov <ksemenov@pivotal.io>

### CODE REVIEW

**Comment by wy65701436** [comment]
thanks @jhvhs, it's very helpful for the new contributor who wants to run test locally, we'll have a review soon.

**Comment by xaleeks** [comment]
@reasonerjt please review this

**Comment by xaleeks** [comment]
@stonezdj  please share what we discussed on why this is not being merged at this time.

**Comment by jhvhs** [comment]
@xaleeks Is there any update for this issue?

File: tests/local/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    libxss1=1:1.2.2-1 locales=2.23-0ubuntu11 mercurial=3.7.3-1ubuntu1.2 net-tools=1.60-26ubuntu1 \
+    netcat=1.10-41 openjdk-9-jdk-headless=9~b114-0ubuntu1 openssh-client=1:7.2p2-4ubuntu2.8 \
+    parallel=20161222-1~ubuntu0.16.04.1 postgresql=9.5+173ubuntu0.2 software-properties-common=0.96.20.9 \
+    sudo=1.8.16-0ubuntu1.7 tar=1.28-2.1ubuntu0.1 unzip=6.0-20ubuntu1 wget=1.17.1-1ubuntu1.5 \
+    xdg-utils=1.1.1-1ubuntu1.16.04.3 xvfb=2:1.18.4-0ubuntu0.8 zip=3.0-11 \
+  && apt-get clean \
+  && rm -rf /var/lib/apt/lists/*
+
+# Set timezone to UTC by default
+RUN ln -sf /usr/share/zoneinfo/Etc/UTC /etc/localtime
```
Comment (bitsf) [inline tests/local/Dockerfile]: seems no such folder /usr/share/zoneinfo/Etc/

File: tests/local/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+RUN add-apt-repository ppa:longsleep/golang-backports
+
+RUN apt-get update && apt-get install -y --no-install-recommends containerd.io=1.2.6-3 \
+    docker-ce=5:19.03.2~3-0~ubuntu-xenial docker-ce-cli=5:19.03.2~3-0~ubuntu-xenial \
+    golang-go=2:1.13~1longsleep1+xenial \
+  && apt-get clean \
+  && rm -rf /var/lib/apt/lists/*
+
+# docker compose
+RUN COMPOSE_URL="https://circle-downloads.s3.amazonaws.com/circleci-images/cache/linux-amd64/docker-compose-latest" \
```
Comment (bitsf) [inline tests/local/Dockerfile]: it's better install docker-compose from
https://github.com/docker/compose/releases/download/1.24.1/docker-compose-$(uname -s)-$(uname -m)

**Comment by bitsf** [comment]
will try this later

**Comment by bitsf** [comment]
We have moved from travis to use github action now.
Github action doesn't have job limitation now.
Github action just use a clean ubuntu as base, and we can use local machine to run the CI test directly now.

So close this PR.
Still thanks a lot for the effort.

**Pull request closed without merging**